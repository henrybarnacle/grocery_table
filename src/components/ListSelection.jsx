import React from 'react';

const ListSelection = ({selected}) => {
  return <div className="listSelection">{selected.name}</div>
  };

export default ListSelection;
