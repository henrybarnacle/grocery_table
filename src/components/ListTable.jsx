import React from 'react';
import useToggle from '../hooks/useToggle';

export const ListTable = (props) => {

  const [darkTheme, setDarkTheme] = useToggle();

  const onSelect = (entry) => { 
    if (props.selected.id === entry.id) {
      props.deselectItem(entry);
    } else {
      props.selectItem(entry);
    }
  }
  return (
    <div className={darkTheme ? 'listTable darkMode' : 'listTable'}>
            <thead>
              <tr>
                  <th>Item</th>
                  <th>Category</th>
                  <th>Delivery</th>
                  <th></th>
              </tr>
            </thead>
            <tbody>
              {props.groceryList.map && props.groceryList.map(entry => {
                  return (
                  <tr key={entry.id}>
                    <td onClick={() => {onSelect(entry)}}>{entry.name}</td>
                    <td>{entry.category}</td>
                    <td>{entry.deliveryMethod}</td>
                    <td>
                    <button className={darkTheme ? 'darkButton' : ''} onClick={() => {onSelect(entry)}}>select</button>
                    <button className={darkTheme ? 'darkButton' : ''} onClick={() => {props.removeItem(entry.id)}}>remove</button>
                    </td>
                  </tr>
                  )
                }
                )}
            </tbody>
            <button onClick={setDarkTheme}>{darkTheme ? 'light table' : 'dark table'}</button>
    </div>
  );
}

export default ListTable
