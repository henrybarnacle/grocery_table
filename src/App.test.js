import { ListContainer } from './components/ListContainer';
import ListSelection from './components/ListSelection';
import ListTable from './components/ListTable';
import ListInputs from './components/ListInputs';
import reducer, { initialState, ADD_ITEM, REMOVE_ITEM } from './ducks/groceries';
import { render, unmountComponentAtNode } from 'react-dom';
import { shallow} from 'enzyme';
import { act } from 'react-dom/test-utils';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const list = initialState.list;
const store = mockStore({
    groceries: {
      list: list,     
          selectedItem: list[3]
        }
});
let container = null;
let wrapped = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
  wrapped = shallow(<ListContainer/>);
});
afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('shows ListSelection component', () => {
  expect(wrapped.find(ListSelection).length).toEqual(1);
});

it('shows ListTable component', () => {
  expect(wrapped.find(ListTable).length).toEqual(1);
});

it('shows ListInputs component', () => {
  expect(wrapped.find(ListInputs).length).toEqual(1);
});

it('handles actions of type REMOVE_ITEM', () => {
  const action = {
    type: REMOVE_ITEM,
    payload: 66
  };
  const newState = reducer(initialState, action);
  expect(newState.list.length).toEqual(3);
});

it('handles actions of type ADD_ITEM', () => {
  const action = {
    type: ADD_ITEM,
    payload: 66
  };
  const newState = reducer(initialState, action);
  expect(newState.list.length).toEqual(5);
});

it('renders name in ListSelection component', () => {
  let item = store.getState().groceries.selectedItem;
  act(() => {
    render(<ListSelection selected={item}/>, container);
  });
  expect(container.textContent).toBe('Roasted Turkey');
});

it('renders list in ListTable component', () => {
  let list = store.getState().groceries.list;
  act(() => {
    render(<ListTable groceryList={list}/>, container);
  });
  expect(container.innerHTML).toContain('Bananas');
  expect(container.innerHTML).toContain('Lettuce');
  expect(container.innerHTML).toContain('Roasted Turkey');
  expect(container.innerHTML).toContain('Whole Grain Bread');
});
