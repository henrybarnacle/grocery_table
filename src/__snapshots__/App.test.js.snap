// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
<section
  className="groceryApp"
>
  <div
    className="listInputs"
  >
    <button
      className="addItemButton"
      onClick={[Function]}
    >
      Add Random Item
    </button>
  </div>
  <div
    className="types"
  >
    <div
      className="listSelection"
    >
      Roasted Turkey
    </div>
    <div
      className="listTable"
    >
      <thead>
        <tr>
          <th>
            Item
          </th>
          <th>
            Category
          </th>
          <th>
            Delivery
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            onClick={[Function]}
          >
            Roasted Turkey
          </td>
          <td>
            Deli
          </td>
          <td>
            Ground
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              remove
            </button>
          </td>
        </tr>
      </tbody>
    </div>
  </div>
</section>
`;

exports[`ListContainer component renders without crashing 1`] = `
<section
  className="groceryApp"
>
  <div
    className="listInputs"
  >
    <button
      className="addItemButton"
      onClick={[Function]}
    >
      Add Random Item
    </button>
  </div>
  <div
    className="types"
  >
    <div
      className="listSelection"
    >
      Roasted Turkey
    </div>
    <div
      className="listTable"
    >
      <thead>
        <tr>
          <th>
            Item
          </th>
          <th>
            Category
          </th>
          <th>
            Delivery
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            onClick={[Function]}
          >
            Roasted Turkey
          </td>
          <td>
            Deli
          </td>
          <td>
            Ground
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              remove
            </button>
          </td>
        </tr>
      </tbody>
    </div>
  </div>
</section>
`;

exports[`ListSelection component renders without crashing 1`] = `
<section
  className="groceryApp"
>
  <div
    className="listInputs"
  >
    <button
      className="addItemButton"
      onClick={[Function]}
    >
      Add Random Item
    </button>
  </div>
  <div
    className="types"
  >
    <div
      className="listSelection"
    >
      Roasted Turkey
    </div>
    <div
      className="listTable"
    >
      <thead>
        <tr>
          <th>
            Item
          </th>
          <th>
            Category
          </th>
          <th>
            Delivery
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            onClick={[Function]}
          >
            Roasted Turkey
          </td>
          <td>
            Deli
          </td>
          <td>
            Ground
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              remove
            </button>
          </td>
        </tr>
      </tbody>
    </div>
  </div>
</section>
`;

exports[`renders ListContainer without crashing 1`] = `
<section
  className="groceryApp"
>
  <div
    className="listInputs"
  >
    <button
      className="addItemButton"
      onClick={[Function]}
    >
      Add Random Item
    </button>
  </div>
  <div
    className="types"
  >
    <div
      className="listSelection"
    >
      Roasted Turkey
    </div>
    <div
      className="listTable"
    >
      <thead>
        <tr>
          <th>
            Item
          </th>
          <th>
            Category
          </th>
          <th>
            Delivery
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            onClick={[Function]}
          >
            Bananas
          </td>
          <td>
            Fruit
          </td>
          <td>
            Air
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              remove
            </button>
          </td>
        </tr>
        <tr>
          <td
            onClick={[Function]}
          >
            Whole Grain Bread
          </td>
          <td>
            Grains
          </td>
          <td>
            Air
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              remove
            </button>
          </td>
        </tr>
        <tr>
          <td
            onClick={[Function]}
          >
            Lettuce
          </td>
          <td>
            Vegetable
          </td>
          <td>
            Ground
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              remove
            </button>
          </td>
        </tr>
        <tr>
          <td
            onClick={[Function]}
          >
            Roasted Turkey
          </td>
          <td>
            Deli
          </td>
          <td>
            Ground
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              remove
            </button>
          </td>
        </tr>
      </tbody>
    </div>
  </div>
</section>
`;

exports[`renders without crashing 1`] = `
<section
  className="groceryApp"
>
  <div
    className="listInputs"
  >
    <button
      className="addItemButton"
      onClick={[Function]}
    >
      Add Random Item
    </button>
  </div>
  <div
    className="types"
  >
    <div
      className="listSelection"
    >
      Roasted Turkey
    </div>
    <div
      className="listTable"
    >
      <thead>
        <tr>
          <th>
            Item
          </th>
          <th>
            Category
          </th>
          <th>
            Delivery
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            onClick={[Function]}
          >
            Bananas
          </td>
          <td>
            Fruit
          </td>
          <td>
            Air
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              remove
            </button>
          </td>
        </tr>
        <tr>
          <td
            onClick={[Function]}
          >
            Whole Grain Bread
          </td>
          <td>
            Grains
          </td>
          <td>
            Air
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              remove
            </button>
          </td>
        </tr>
        <tr>
          <td
            onClick={[Function]}
          >
            Lettuce
          </td>
          <td>
            Vegetable
          </td>
          <td>
            Ground
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              remove
            </button>
          </td>
        </tr>
        <tr>
          <td
            onClick={[Function]}
          >
            Roasted Turkey
          </td>
          <td>
            Deli
          </td>
          <td>
            Ground
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              remove
            </button>
          </td>
        </tr>
      </tbody>
    </div>
  </div>
</section>
`;
